charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -90,23 +170,130 @@ readOnlyRootFilesystem: true
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Helper to toggle the creation of the job that creates and registers the system identity.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.shouldRunJob" -}}
+{{- $privateKey := include "newrelic-super-agent.auth.secret.privateKey.data" . -}}
+{{- $clientId := include "newrelic-super-agent.auth.secret.clientId.data" . -}}
+
+{{- if and ((.Values.config).auth).enabled (((.Values.config).auth).secret).create (not $privateKey) (not $clientId) -}}
+  true
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Helper to toggle the creation of the secret that has the system identity as values.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.shouldTemplate" -}}
+{{- if and ((.Values.config).auth).enabled (((.Values.config).auth).secret).create -}}
+  {{- $privateKey := include "newrelic-super-agent.auth.secret.privateKey.data" . -}}
+  {{- $clientId := include "newrelic-super-agent.auth.secret.clientId.data" . -}}
+
+  {{- if and $privateKey $clientId -}}
+    true
+  {{- else if or $privateKey $clientId -}}
+    {{- fail "If you provide your own system identity data you have to provide both private key and client id" -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Check if .Values.config.auth.secret.private_key.secret_key exists and use it for the key in the secret containing the private
+key needed for the system identity. Fallbacks to `private_key`.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.privateKey.key" -}}
+{{- $key := (((((.Values.config).auth).secret).private_key).secret_key) -}}
+{{- if $key -}}
+  {{- $key -}}
+{{- else -}}
+  private_key
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Check if .Values.config.auth.secret.private_key.(plain_pem or base64_pem) exists and use it for as the private certificate for
+auth. If no ceritifcate is provided, it defaults to `""` (empty string) so this helper can be used directly as a test.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.privateKey.data" -}}
+{{- $plain_pem := (((((.Values.config).auth).secret).private_key).plain_pem) -}}
+{{- $base64_pem := (((((.Values.config).auth).secret).private_key).base64_pem) -}}
+{{- if and $plain_pem $base64_pem -}}
+  {{- fail "Only one of base64_pem or plain_pem should be provided it you want to provide your own certificate." -}}
+{{- else if $base64_pem -}}
+  {{- $base64_pem }}
+{{- else if $plain_pem -}}
+  {{- $plain_pem | b64enc -}}
+{{- else -}}
+  {{- /* Empty string */ -}}
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Check if .Values.config.auth.secret.client_id.secret_key exists and use it for the key in the secret containing the client id
+needed for the system identity. Fallbacks to `client_id`.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.clientId.key" -}}
+{{- $key := (((((.Values.config).auth).secret).client_id).secret_key) -}}
+{{- if $key -}}
+  {{- $key -}}
+{{- else -}}
+  CLIENT_ID
```

Comment:
I am not sure if it worth to make these secret keys configurable, the work is already done, but maybe it would have been easier to just force the Key name

Reply:
There are people that want a secret per license and people that want one secret for all licenses.

We have to make it configurable to cover both use cases.